src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,161 @@
+import { useState } from 'react';
+import { updateTodos } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+
+import cn from 'classnames';
+import { useFocusInput } from '../../hooks/useFocusInput';
+
+type Props = {
+  todo: Todo;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  isLoading?: boolean;
+  handleDeleteTodo: (id: number) => void;
+  isDeleting?: boolean;
+  setloadingIds: (
+    loading: number[] | ((prevLoading: number[]) => number[]),
+  ) => void;
+  loadingIds: number[];
+  setErrorMessage: (string: string) => void;
+  handleUpdateTodo: (todo: Todo) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  setTodos,
+  isLoading,
+  handleDeleteTodo,
+  isDeleting,
+  setloadingIds,
+  loadingIds,
+  setErrorMessage,
+  handleUpdateTodo,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+  const focusOnInput = useFocusInput();
+
+  function toggleCompleteTodo() {
+    setloadingIds(prevIds => [...prevIds, todo.id]);
+
+    const updatedTodo = { ...todo, completed: !todo.completed };
+
+    updateTodos(todo.id, updatedTodo)
+      .then(() => {
+        setTodos((prevTodos: Todo[]) =>
+          prevTodos.map(todoItem =>
+            todoItem.id === todo.id ? updatedTodo : todoItem,
+          ),
+        );
+      })
+      .catch(() => setErrorMessage('Unable to update a todo'))
+      .finally(() => {
+        setloadingIds(prevIds => prevIds.filter(id => id !== todo.id));
+      });
+  }
+
+  const editTodo = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const trimmedTitle = newTitle.trim();
+
+    if (!trimmedTitle) {
+      handleDeleteTodo(todo.id);
+
+      return;
+    }
+
+    if (trimmedTitle === todo.title) {
+      setIsEditing(false);
+
+      return;
+    }
+
+    try {
+      await handleUpdateTodo({
+        ...todo,
+        title: trimmedTitle,
+      });
+      setIsEditing(false);
+    } catch (error) {
+      setIsEditing(false);
+    }
```

Comment:
```suggestion
      await handleUpdateTodo({
        ...todo,
        title: trimmedTitle,
      }, onSuccess);
```

onSucces can be function which will to do somethin after success response.
for example you need to reset editing only on success response.
```
const onSuccess = () => {
   setIsEditing(false);
   // maybe something else todo after success response
}
```